### ISSUE TITLE
BUG: groupby.diff(axis="index") puts keys in index but groupby.diff(axis=0) does not

### ISSUE THREAD

**Issue by mvashishtha** [pandas-dev/pandas#53688]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
print(pd.DataFrame([[1, 2]]).groupby(0).diff(axis='index'))
print(pd.DataFrame([[1, 2]]).groupby(0).diff(axis=0))
```

### Issue Description

groupby.diff(axis="index") puts keys in index but groupby.diff(axis=0) does not

### Expected Behavior

Based on warnings from older versions of pandas, I think the `axis="index"` behavior is expected, although I do think it's weird that the index seems to include both the original index and the group keys, so in this example it has two levels.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 965ceca9fd796940050d6fc817707bba1c4f9bff
python           : 3.8.16.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.5.0
Version          : Darwin Kernel Version 22.5.0: Mon Apr 24 20:51:50 PDT 2023; root:xnu-8796.121.2~5/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.2
numpy            : 1.24.3
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.8.0
pip              : 23.0.1
Cython           : None
pytest           : 7.3.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 8.12.2
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2023.6.0
gcsfs            : None
matplotlib       : 3.7.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 10.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

**Comment by rhshadrach** [comment]
Thanks for the report! This is fixed on main, and the axis argument is deprecated, so I think we're okay closing this without tests.

> Based on warnings from older versions of pandas, I think the `axis="index"` behavior is expected, although I do think it's weird that the index seems to include both the original index and the group keys, so in this example it has two levels.

In pandas, transformations (of which `diff` is an example) return the index of the original object. So the result without the keys in the index is the "correct" one. See https://pandas.pydata.org/pandas-docs/dev/user_guide/groupby.html#transformation for some details.

**Comment by mvashishtha** [comment]
@rhshadrach thank you for the response. I see that this has indeed been fixed on commit 870a504af96a2ebcb89f2b09b110a9358f7771e7
